src/ADAL.PCL/ADAL.PCL.csproj

```diff
@@ -60,84 +60,85 @@
     <Compile Include="..\ADAL.Common\CommonAssemblyInfo.cs">
       <Link>CommonAssemblyInfo.cs</Link>
     </Compile>
-    <Compile Include="AcquireDeviceCodeHandler.cs" />
-    <Compile Include="AcquireTokenByAuthorizationCodeHandler.cs" />
-    <Compile Include="AcquireTokenByDeviceCodeHandler.cs" />
-    <Compile Include="AcquireTokenForClientHandler.cs" />
-    <Compile Include="AcquireTokenHandlerBase.cs" />
-    <Compile Include="AcquireTokenInteractiveHandler.cs" />
-    <Compile Include="AcquireTokenNonInteractiveHandler.cs" />
-    <Compile Include="AcquireTokenOnBehalfHandler.cs" />
-    <Compile Include="AcquireTokenSilentHandler.cs" />
-    <Compile Include="AdalException.cs" />
-    <Compile Include="AdalIdHelper.cs" />
-    <Compile Include="AdalServiceException.cs" />
-    <Compile Include="AdalSilentTokenAcquisitionException.cs" />
-    <Compile Include="AdalUserMismatchException.cs" />
+    <Compile Include="Flows\DeviceCode\AcquireDeviceCodeHandler.cs" />
+    <Compile Include="Flows\AcquireTokenByAuthorizationCodeHandler.cs" />
+    <Compile Include="Flows\DeviceCode\AcquireTokenByDeviceCodeHandler.cs" />
+    <Compile Include="Flows\AcquireTokenForClientHandler.cs" />
+    <Compile Include="Flows\AcquireTokenHandlerBase.cs" />
+    <Compile Include="Flows\AcquireTokenInteractiveHandler.cs" />
+    <Compile Include="Flows\NonInteractive\AcquireTokenNonInteractiveHandler.cs" />
+    <Compile Include="Flows\AcquireTokenOnBehalfHandler.cs" />
+    <Compile Include="Flows\AcquireTokenSilentHandler.cs" />
+    <Compile Include="Exceptions\AdalException.cs" />
+    <Compile Include="Utilities\AdalIdHelper.cs" />
+    <Compile Include="Exceptions\AdalServiceException.cs" />
+    <Compile Include="Exceptions\AdalSilentTokenAcquisitionException.cs" />
+    <Compile Include="Exceptions\AdalUserMismatchException.cs" />
     <Compile Include="AuthenticationContext.cs" />
-    <Compile Include="AuthenticationParameters.cs" />
-    <Compile Include="AuthenticationResult.cs" />
-    <Compile Include="Authenticator.cs" />
-    <Compile Include="AuthenticatorTemplate.cs" />
+    <Compile Include="Data\AuthenticationParameters.cs" />
+    <Compile Include="Data\AuthenticationResult.cs" />
+    <Compile Include="Data\Authenticator.cs" />
+    <Compile Include="Data\AuthenticatorTemplate.cs" />
```

Comment:
consider renaming /Data to /Authority. Also, authenticantionparameters and authenticationresult do not belong to the /Data.